azurerm/resource_arm_private_endpoint.go

```diff
@@ -0,0 +1,380 @@
+package azurerm
+
+import (
+	"fmt"
+	"log"
+
+	"github.com/Azure/azure-sdk-for-go/services/network/mgmt/2019-06-01/network"
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/helpers/azure"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/helpers/response"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/helpers/tf"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/helpers/validate"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/features"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/tags"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/utils"
+)
+
+func resourceArmPrivateEndpoint() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceArmPrivateEndpointCreateUpdate,
+		Read:   resourceArmPrivateEndpointRead,
+		Update: resourceArmPrivateEndpointCreateUpdate,
+		Delete: resourceArmPrivateEndpointDelete,
+
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validate.NoEmptyStrings,
```

Comment:
Not a blocker but ideally i'd like a regex to validate the name against here (and the other name bits of this schema)

Reply:
Yep... I have that code in the other [PR](https://github.com/terraform-providers/terraform-provider-azurerm/pull/4426) once that is merged I can reference the helper functions for validation here.